mypy/test/meta/test_diff_helper.py

```diff
@@ -0,0 +1,47 @@
+import io
+
+from mypy.test.helpers import Suite, diff_ranges, render_diff_range
+
+
+class DiffHelperSuite(Suite):
+    def test_render_diff_range(self) -> None:
+        expected = ["hello", "world"]
+        actual = ["goodbye", "world"]
+
+        expected_ranges, actual_ranges = diff_ranges(expected, actual)
+
+        output = io.StringIO()
+        render_diff_range(expected_ranges, expected, output=output)
+        assert output.getvalue() == "  hello (diff)\n  world\n"
+        output = io.StringIO()
+        render_diff_range(actual_ranges, actual, output=output)
+        assert output.getvalue() == "  goodbye (diff)\n  world\n"
+
+        expected = ["a", "b", "c", "d", "e", "f", "g", "h", "circle", "i", "j"]
+        actual = ["a", "b", "c", "d", "e", "f", "g", "h", "square", "i", "j"]
+
+        expected_ranges, actual_ranges = diff_ranges(expected, actual)
+
+        output = io.StringIO()
+        render_diff_range(expected_ranges, expected, output=output, indent=0)
```

Comment:
I'veb lately acquired a taste for the arrange-act-assert style. Might be a good fit here given how similar the two methods are named, and how you're using testing terminology ("expected") in a meta fashion.

Reply:
Wasn't aware of that term. I think I have a little blurring between act and assert, but maybe that's okay? I am asserting two different things with similar "arrangement"

Reply:
As a general rule, "# Act" is where you call the function under test. I think it's ok to assert multiple things, probably just try not to eg. define expectations ahead of the "# Assert"  section.